google_cloud_automlops/AutoMLOps.py

```diff
@@ -0,0 +1,546 @@
+    orchestration_framework: Optional[str] = Orchestrator.KFP.value,
+    pipeline_job_runner_service_account: Optional[str] = None,
+    pipeline_job_submission_service_location: Optional[str] = DEFAULT_RESOURCE_LOCATION,
+    pipeline_job_submission_service_name: Optional[str] = None,
+    pipeline_job_submission_service_type: Optional[str] = PipelineJobSubmitter.CLOUD_FUNCTIONS.value,
+    precheck: Optional[bool] = True,
+    provision_credentials_key: str = None,
+    provisioning_framework: Optional[str] = Provisioner.GCLOUD.value,
+    pubsub_topic_name: Optional[str] = None,
+    schedule_location: Optional[str] = DEFAULT_RESOURCE_LOCATION,
+    schedule_name: Optional[str] = None,
+    schedule_pattern: Optional[str] = DEFAULT_SCHEDULE_PATTERN,
+    source_repo_branch: Optional[str] = DEFAULT_SOURCE_REPO_BRANCH,
+    source_repo_name: Optional[str] = None,
+    source_repo_type: Optional[str] = CodeRepository.CLOUD_SOURCE_REPOSITORIES.value,
+    storage_bucket_location: Optional[str] = DEFAULT_RESOURCE_LOCATION,
+    storage_bucket_name: Optional[str] = None,
+    hide_warnings: Optional[bool] = True,
+    use_ci: Optional[bool] = False,
+    vpc_connector: Optional[str] = DEFAULT_VPC_CONNECTOR):
+    """Generates relevant pipeline and component artifacts,
+       then provisions resources, builds, compiles, and submits the PipelineJob.
+       Check constants file for variable default values.
+
+    Args:
+        project_id: The project ID.
+        pipeline_params: Dictionary containing runtime pipeline parameters.
+        artifact_repo_location: Region of the artifact repo (default use with Artifact Registry).
+        artifact_repo_name: Artifact repo name where components are stored (default use with Artifact Registry).
+        artifact_repo_type: The type of artifact repository to use (e.g. Artifact Registry, JFrog, etc.)        
+        base_image: The image to use in the component base dockerfile.
+        build_trigger_location: The location of the build trigger (for cloud build).
+        build_trigger_name: The name of the build trigger (for cloud build).
+        custom_training_job_specs: Specifies the specs to run the training job with.
+        deployment_framework: The CI tool to use (e.g. cloud build, github actions, etc.)
+        naming_prefix: Unique value used to differentiate pipelines and services across AutoMLOps runs.
+        orchestration_framework: The orchestration framework to use (e.g. kfp, tfx, etc.)
+        pipeline_job_runner_service_account: Service Account to run PipelineJobs.
+        pipeline_job_submission_service_location: The location of the cloud submission service.
+        pipeline_job_submission_service_name: The name of the cloud submission service.
+        pipeline_job_submission_service_type: The tool to host for the cloud submission service (e.g. cloud run, cloud functions).
+        precheck: Boolean used to specify whether to check for provisioned resources before deploying.
```

Comment:
Confused about why this is a parameter? What situation would not benefit from a precheck?

Reply:
The precheck functionality is somewhat heavyweight in that it has to use the discovery service to query API endpoints and determine if certain resources exist. This incurs additional costs and can tax the endpoints; basically if the user is confident that the resources exist because they've run it many times in the same project, then they can set precheck to false

Reply:
Is it only really used for gcloud provisioning? I think my other big question is, if we have separated into provision generate, etc., shouldn't the confidence in resource existence be evident by just not running the provision function again? In terms of use case for the launch all function, precheck = True would just mean they don't want to run the provision portion. If this is the case I think we should be almost dissuading them from relying on this launch all and push towards using the specific functions. Or, phrased differently, I would think that the people using launch all are perhaps those who are less comfortable with the whole process, and want a one click experience. For those people, I would think they are less experienced and would want to use precheck there.

Just food for thought though, don't feel super strongly either way. I'm always just looking for a way to use less parameters.